docs: Module 3 IBC, Bridges (B9lab updates 2-6) (#810)
cosmos/sdk-tutorials: b9lab-content/3-main-concepts/16-ibc.md

## Modular design

IBC is an end-to-end, connection-oriented, stateful protocol for reliable, ordered, and authenticated communication between heterogeneous blockchains arranged in an unknown and dynamic topology.

This is accomplished by specifying a set of data structures, abstractions, and semantics that can be implemented by any distributed ledger provided they satisfy a small set of requirements.

IBC can be used to build a wide range of cross-chain applications, which include token transfers, atomic swaps, multi-chain smart contracts (with or without mutually comprehensible virtual machines, and data & code sharding of various kinds.

Modules do not require in-depth knowledge of the low-level details of clients, connections, and proof verification.

## Components

<HighlightBox type=”info”>
Components https://github.com/cosmos/cosmos-sdk/blob/master/docs/ibc/overview.md 
</HighlightBox>

### Clients

IBC clients are light clients that are identified by a unique client ID. IBC clients track the consensus states of other blockchains and the proof specs of those blockchains that are required to properly verify proofs against the client's consensus state.

> A client can be associated with any number of connections to multiple chains

Reviewer: ```suggestion
A client can be associated with any number of connections to one chain
```

### Acknowledgement

Modules also write application-specific acknowledgements when processing a packet Acknowledgements can be done:

- Synchronously on `OnRecvPacket` if the module processes packets as soon as they are received from IBC module
* Asynchronously if the module processes packets at some later point after receiving the packet

This acknowledgement data is opaque to IBC much like the packet Data and is treated by IBC as a simple byte string `[]byte`.

Receiver modules must encode their acknowledgement so that the sender module can decode it correctly. How the acknowledgement is encoded should be decided through version negotiation during the channel handshake.

After the acknowledgement has been written by the receiving chain, a relayer relays the acknowledgement back to the original sender module which then executes application-specific acknowledgment logic using the contents of the acknowledgement.

When the acknowledgement is received successfully on the original sender chain, the IBC module deletes the corresponding packet commitment as it is no longer needed

## Types of chain

> IBC can use Tendermint chains to bridge with IBC but also non-Tendermint chains. Two types of non-Tendermint chains are supported:

Reviewer: ```suggestion
Chains using the Tendermint consensus algorithm can bridge using IBC, but also non-Tendermint chains can use IBC. Two types of non-Tendermint chains are supported:
```
